lib/couchbase/protostellar/connect_options.rb

```diff
@@ -35,13 +35,17 @@ def grpc_credentials
         end
       end
 
-      def grpc_channel_args
+      def grpc_call_metadata
         if @username && @password
-          {"authorization" => "Basic #{Base64.encode64("#{@username}:#{@password}")}"}
+          {"authorization" => ["Basic #{Base64.encode64("#{@username}:#{@password}")}".chop]}
```

Comment:
The `chop` here is because `Base64.encode64` adds a new line at the end and GRPC doesn't like that

Reply:
And in fact `chop`, `chomp` or `strip` won't help here, because `encode64` might insert line feed characters if the input is long enough
```ruby
Base64.encode64("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx")
#=> "eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4\neHh4eHh4eA==\n"
```